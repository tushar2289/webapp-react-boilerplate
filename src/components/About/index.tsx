import React from 'react';
import Paper from '@material-ui/core/Paper';
import Link from '@material-ui/core/Link';
import makeStyles from '@material-ui/core/styles/makeStyles';
import createStyles from '@material-ui/core/styles/createStyles';
import { Theme } from '@material-ui/core/styles';
import Box from '@material-ui/core/Box';
import Container from '@material-ui/core/Container';
import Typography from '@material-ui/core/Typography';
import GitHubIcon from '@material-ui/icons/GitHub';
import FavoriteIcon from '@material-ui/icons/Favorite';
import Logo from '../../logo.svg';
import {
  GITHUB_LINK,
  MOBX_LINK,
  MATERIAL_UI_LINK,
  AXIOS_LINK,
  TYPESCRIPT_LINK,
  REACT_LINK,
} from '../../constants';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      margin: theme.spacing(4),
      textAlign: 'center',
    },
    logo: {
      width: '200px',
    },
  })
);

const About: React.FC = () => {
  const classes = useStyles();
  const preventDefault = (event: React.SyntheticEvent) =>
    event.preventDefault();
  return (
    <Container maxWidth="sm">
      <Paper square className={classes.root}>
        <img src={Logo} className={classes.logo} alt="logo" />
        <Box
          display="flex"
          alignItems="center"
          justifyContent="center"
          marginBottom={1}
        >
          <GitHubIcon />
          <Box paddingLeft={1}>
            <Link
              color="textPrimary"
              href={GITHUB_LINK}
              onClick={preventDefault}
              variant="h6"
            >
              Source code
            </Link>
          </Box>
        </Box>
        <Box paddingBottom={1}>
          <Typography variant="caption">
            This project was built using the following:
          </Typography>
          <Link
            color="textPrimary"
            href={REACT_LINK}
            onClick={preventDefault}
            variant="h6"
            display="block"
          >
            React
          </Link>
          <Link
            color="textPrimary"
            href={MOBX_LINK}
            onClick={preventDefault}
            variant="h6"
            display="block"
          >
            MobX
          </Link>
          <Link
            color="textPrimary"
            href={MATERIAL_UI_LINK}
            onClick={preventDefault}
            variant="h6"
            display="block"
          >
            Material UI
          </Link>
          <Link
            color="textPrimary"
            href={AXIOS_LINK}
            onClick={preventDefault}
            variant="h6"
            display="block"
          >
            Axios
          </Link>
          <Link
            color="textPrimary"
            href={TYPESCRIPT_LINK}
            onClick={preventDefault}
            variant="h6"
            display="block"
          >
            Typescript
          </Link>
        </Box>
        <Box padding={2}>
          <Typography variant="h5">
            Made with <FavoriteIcon color="error" /> in India
          </Typography>
        </Box>
      </Paper>
    </Container>
  );
};

export default About;
