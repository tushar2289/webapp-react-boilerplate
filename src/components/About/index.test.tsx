import React from 'react';
import { render, cleanup } from '@testing-library/react';
import About from './index';

afterEach(cleanup);

test('renders', () => {
  const { asFragment } = render(<About />);
  expect(asFragment()).toMatchSnapshot();
});

test('renders github link', () => {
  const { getByText } = render(<About />);
  expect(getByText('Source code')).toBeInTheDocument();
});
