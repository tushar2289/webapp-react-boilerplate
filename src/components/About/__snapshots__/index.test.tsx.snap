// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthSm"
  >
    <div
      class="MuiPaper-root makeStyles-root-1 MuiPaper-elevation1"
    >
      <img
        alt="logo"
        class="makeStyles-logo-2"
        src="logo.svg"
      />
      <div
        class="MuiBox-root MuiBox-root-3"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
          />
        </svg>
        <div
          class="MuiBox-root MuiBox-root-4"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-h6 MuiTypography-colorTextPrimary"
            href="https://github.com/tushar2289/webapp-react-boilerplate"
          >
            Source code
          </a>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-5"
      >
        <span
          class="MuiTypography-root MuiTypography-caption"
        >
          This project was built using the following:
        </span>
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-h6 MuiTypography-colorTextPrimary MuiTypography-displayBlock"
          href="https://reactjs.org/"
        >
          React
        </a>
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-h6 MuiTypography-colorTextPrimary MuiTypography-displayBlock"
          href="https://mobx-state-tree.js.org/intro/philosophy"
        >
          MobX
        </a>
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-h6 MuiTypography-colorTextPrimary MuiTypography-displayBlock"
          href="https://material-ui.com/"
        >
          Material UI
        </a>
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-h6 MuiTypography-colorTextPrimary MuiTypography-displayBlock"
          href="https://github.com/axios/axios"
        >
          Axios
        </a>
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-h6 MuiTypography-colorTextPrimary MuiTypography-displayBlock"
          href="https://www.typescriptlang.org/"
        >
          Typescript
        </a>
      </div>
      <div
        class="MuiBox-root MuiBox-root-6"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
        >
          Made with 
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorError"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
            />
          </svg>
           in India
        </h5>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
